src/utils/formContext.ts

```diff
@@ -2,55 +2,83 @@ import { changeStep, FieldOptions } from './formHelperFunctions';
-    getErrorCallback: (
-      props1?: Record<string, unknown>
-    ) => (props2?: Record<string, unknown>) => Promise<void>;
-    history: any;
-    setInlineErrors: React.Dispatch<
-      React.SetStateAction<Record<string, { message: string; index: number }>>
-    >;
-    setUserProgress: React.Dispatch<React.SetStateAction<null>>;
-    steps: any;
-    updateFieldOptions: (
-      stepData: any,
-      loadStep?: null
-    ) => (newOptions: FieldOptions) => void;
-  }
-) => ({
+export type ContextProps = {
+  client: any;
+  formName: string;
+  formRef: React.MutableRefObject<any>;
+  formSettings: any;
+  getErrorCallback: (
+    props1?: Record<string, unknown>
+  ) => (props2?: Record<string, unknown>) => Promise<void>;
+  history: any;
+  setInlineErrors: React.Dispatch<
+    React.SetStateAction<Record<string, { message: string; index: number }>>
+  >;
+  setUserProgress: React.Dispatch<React.SetStateAction<null>>;
+  steps: any;
+  updateFieldOptions: (
+    stepData: any,
+    loadStep?: null
+  ) => (newOptions: FieldOptions) => void;
+};
+
+export const getFormContext = (formUuid: string) => ({
   setValues,
-  setFormCompletion: (flag: boolean) =>
-    props.client.registerEvent({
-      step_key: newStep.key,
+  setFormCompletion: (flag: boolean) => {
+    const { contextProps, currentStep } = initInfo().formState[formUuid];
+    if (!contextProps) return;
+
+    return contextProps.client.registerEvent({
+      step_key: currentStep.key,
       event: 'load',
       completed: flag
-    }),
-  setOptions: props.updateFieldOptions(props.steps),
-  setProgress: (val: any) => props.setUserProgress(val),
+    });
+  },
+  setOptions: (newOptions: FieldOptions) => {
+    const contextProps = initInfo().formState[formUuid].contextProps;
+    if (!contextProps) return;
+    return contextProps.updateFieldOptions(contextProps.steps)(newOptions);
+  },
+  setProgress: (val: any) => {
+    const contextProps = initInfo().formState[formUuid].contextProps;
+    if (!contextProps) return;
+    return contextProps.setUserProgress(val);
+  },
   setStep: (stepKey: any) => {
-    changeStep(stepKey, newStep.key, props.steps, props.history);
+    console.log('setStep', initInfo().formState);
+    console.log('passed in form uuid', formUuid);
+    const { contextProps, currentStep } = initInfo().formState[formUuid];
+    if (!contextProps) return;
+    changeStep(
+      stepKey,
+      currentStep.key,
+      contextProps.steps,
+      contextProps.history
+    );
   },
   step: {
     style: {
-      backgroundColor: newStep?.default_background_color
+      backgroundColor:
+        // This gets baked in.. and isn't present in the context as a result
+        initInfo().formState[formUuid]?.currentStep?.default_background_color
     }
   },
+  // This gets baked in..
   userId: initInfo().userId,
-  stepName: newStep?.key ?? '',
+  // This gets baked in..
+  stepName: initInfo().formState[formUuid]?.currentStep?.key ?? '',
```

Comment:
add getAttributes fn to context. keep step & stepName for now - mark to deprecation 

update docs for getAttributes